app/api/groups/update_permissions.go

```diff
@@ -141,23 +127,33 @@ func (srv *Service) updateGroupItem(w http.ResponseWriter, r *http.Request) serv
 	return service.NoError
 }
 
-func saveGroupItemDataIntoDB(groupID, itemID, creatorUserGroupID int64, data *formdata.FormData, s *database.DataStore) {
-	dbMap := data.ConstructMapForDB()
-	groupItemScope := s.GroupItems().
-		Where("group_id = ?", groupID).
-		Where("item_id = ?", itemID)
-	found, err := groupItemScope.WithWriteLock().HasRows()
+func checkUserHasRightsToSetCanView(canView string, s *database.DataStore, user *database.User, itemID int64) bool {
+	requiredPermission := canView
+	if requiredPermission == "info" { // no "info" in can_grant_view
+		requiredPermission = "content"
+	}
+	// permissions_generated.can_grant_view_value should be >= data["can_view"]
```

Comment:
can_grant_view_value -> can_grant_view_generated_value. I would even say can_grant_view_generated as this is what is important.

Reply:
fixed